### PR TITLE
KOGITO-879 KOGITO-880 KOGITO-881: State Control API (undo/redo)

### PR DESCRIPTION
**Pull request by pefernan**
KOGITO-879: Define State Control API
KOGITO-880: Create JSInterop layer for StateControl API
KOGITO-881: Integrate State Control API in Stunner to handle Undo & Redo

I had to add two new modules:
- `appformer-kogito-api`: to keep the base kogito APIs (for now only StateControl APIs are there), since this module may grow and the state control is a development linked to the envelope, I thoght it was a good idea to have it there.

- `kie-wb-common-stunner-kogito-runtime-common`: to keep the shared classes nedded for the stunner editor runtimes. Other modules under kie-wb-common-stunner-kogito are also present on BC.

@ederign @tiagobento @jomarko would you mind take a look?

This is part of an ensemble, please merge with:
https://github.com/kiegroup/kogito-tooling/pull/66
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1195
https://github.com/kiegroup/appformer/pull/894
https://github.com/kiegroup/kie-wb-common/pull/3156
https://github.com/kiegroup/jbpm-wb/pull/1414
https://github.com/kiegroup/drools-wb/pull/1310
https://github.com/kiegroup/kie-wb-distributions/pull/1013

### CODE REVIEW

**Comment by pefernan** [comment]
@jomarko could you please review this one too?

File: kie-bom/pom.xml [hunk 1]
```diff
@@ -2439,6 +2439,19 @@
         <type>test-jar</type>
       </dependency>
 
+      <dependency>
+        <groupId>org.kie.workbench.stunner</groupId>
+        <artifactId>kie-wb-common-stunner-kogito-runtime-common</artifactId>
```
Comment (jomarko) [inline kie-bom/pom.xml]: I spotted somewhere Roger asked not introduce this module, but reuse an existing one. 

Just checking if that was discussed more and this PR is correct or this PR will change.

Comment (pefernan) [reply kie-bom/pom.xml]: I'll remove it and move everything into the module Roger suggested, don't worry ;)

Comment (pefernan) [reply kie-bom/pom.xml]: @jomarko after talking to @romartin we won't remove this module for now, having it is a pain but looks like the best solution for now.

**Comment by pefernan** [comment]
@jesuino would you mind take a look?

**Pull request merged**